linkerd/app/core/src/svc.rs

```diff
@@ -185,6 +185,10 @@ impl<S> Stack<S> {
         self.push(stack::MapTargetLayer::new(map))
     }
 
+    pub fn push_map_response<M: Clone>(self, map: M) -> Stack<stack::MapResponse<S, M>> {
```

Comment:
It would be nice if we could bound `M` here, but I'm guessing that would mean introducing a bunch of other bounds so we can know the response type, which is less nice?

Reply:
Yeah. Practically, i've been adding type constraints when i debug compilation issues but remove them before merging, since, for instance, parameterizing on the target type or response (body) makes the layer useful in only a single stack context.